ironfish/src/wallet/wallet.ts

```diff
@@ -394,9 +394,32 @@ export class Wallet {
     return decryptedNotes
   }
 
+  async addPendingTransaction(
+    transaction: Transaction,
+    submittedSequence: number | null,
+  ): Promise<void> {
+    const accounts = []
+    for (const account of this.listAccounts()) {
+      if (!(await account.hasTransaction(transaction.hash()))) {
+        accounts.push(account)
+      }
+    }
+
+    if (accounts.length === 0) {
+      return
+    }
+
+    const decryptedNotesByAccountId = await this.decryptNotes(transaction, null, accounts)
+
+    for (const [accountId, decryptedNotes] of decryptedNotesByAccountId) {
+      const account = this.accounts.get(accountId)
+      Assert.isNotUndefined(account, `addPendingTransaction: No account found for ${accountId}`)
```

Comment:
This is a code bomb. Decryption is slow, if you remove an account it's likely to crash if its in the middle of adding a pending transaction and that gossip is for the account being removed.

Reply:
Should we `continue` if the account is removed during decryption?

We have the same check in `syncTransaction`